### PR TITLE
Implemented FormData parsing in curl_parser and added tests for it.

### LINKED ISSUE #503: [curl_parser] Improvement to the way curl parsing handles form-data
The current implementation of the Curl class does not correctly handle form data (-F flag) in cURL commands. Specifically, multipart form data is not being parsed or included in the resulting Curl object, which limits support for requests requiring file uploads or complex form submissions. This issue affects the ability to fully represent cURL commands in Dart

I was thinking of improving the parsing by allowing multiple -F flags and then returning an additional formFields Map<String,dynamic> but this will have an issue. As you might know sending out lists in formdata we are required to do it like 

image

image

So if we like to parse the curl it will be having curl .... -F "arr=ele1" -F "arr=ele2"

which should result in a map of arr = [ele1,ele2]

Hence we wouldn't be able to return formData as a Map because of the non-unique key therefore I think a list of tuples will be better or a list of formdata class objects

This can be achieved easily I am just writing up my approach.

Suggestions?

### PR DESCRIPTION
**Pull request by WrathOP**
## PR Description

_Add your description_

## Related Issues

- Closes #503 

### Checklist
- [x] I have gone through the [contributing guide](https://github.com/foss42/apidash/blob/main/CONTRIBUTING.md)
- [x] I have updated my branch and synced it with project `main` branch before making this PR
- [x] I am using the latest Flutter stable branch (run `flutter upgrade` and verify)
- [x] I have run the tests (`flutter test`) and all tests are passing

## Added/updated tests?
_We encourage you to add relevant test cases._

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests have not been included_

## OS on which you have developed and tested the feature?

- [ ] Windows
- [x] macOS
- [ ] Linux

### CODE REVIEW

**Comment by ashitaprasad** [comment]
@WrathOP can you segregate the tests into three files:
- `curl_model_test.dart` -> Where you test curl model files are getting creating properly
- `curl_parser_test.dart` -> where curl strings are parsed and checked if correct curl model is being obtained.
- `dart_to_curl_test.dart` -> where curl models are converted into curl string and checked if the correct curl command string is obtained.

**Comment by WrathOP** [comment]
Ok sure will do that.

**Comment by WrathOP** [comment]
@ashitaprasad I have split up the files.

File: packages/curl_parser/test/dart_to_curl_test.dart [hunk 1]
```diff
@@ -0,0 +1,218 @@
+import 'package:apidash_core/consts.dart';
```
Comment (ashitaprasad) [inline packages/curl_parser/test/dart_to_curl_test.dart:1]: Have you manually checked if each and every generated string in this file is correct?

File: packages/curl_parser/test/curl_parser_test.dart [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import 'dart:io';
```
Comment (ashitaprasad) [inline packages/curl_parser/test/curl_parser_test.dart:1]: Have you manually checked if each and every generated Curl model in this file is correct?

File: packages/curl_parser/test/dart_to_curl_test.dart (hunk 1 above)
Comment (WrathOP) [reply packages/curl_parser/test/dart_to_curl_test.dart:1]: Thanks for catching that, they were not a valid curl to begin with fixed that. BTW I am using postman as my curl test I hope that it should be good.

**Comment by ashitaprasad** [comment]
Thanks for the PR @WrathOP 🚀

**Pull request merged**